src/brine_nif.erl

```diff
@@ -33,7 +33,11 @@
 -define(nif_error, erlang:nif_error(not_loaded)).
 
 init() ->
-    Workers = erlang:max(1, erlang:round(erlang:system_info(logical_processors_available) * 0.5)),
+    NoOfProcs = case os:type() of
```

Comment:
I think I'd prefer a separate function, `get_worker_count/0`, and it should probably not look at `os:type/0`, but just fall back to `logical_processors` if `logical_processors_available` returns `undefined`. For all we know, it might not work on Solaris either.